pkg/telemetry/phonehome/client_config.go

```diff
@@ -19,6 +24,12 @@ var (
 // services tenant ID. The value of the label becomes the group ID if not empty.
 const TenantIDLabel = "rhacs.redhat.com/tenant"
 
+// Interceptor is a function which will be called on every API call if none of
+// the previous interceptors in the chain returned false.
+// An Interceptor function may add custom properties to the props map so that
+// they appear in the event.
+type Interceptor func(rp *RequestParams, props map[string]any) bool
```

Comment:
Looks like `props` is only expected to return values,, any reason it's passed as a function param?

Reply:
The idea was that a chain of interceptors (per event type) could augment this map one after another. For example, there could be a filtering interceptor that would filter the requests and possibly add some generic properties, and the second one would add more specific properties.
Originally, I was thinking about adding the current `roxctl` event as the second interceptor for `API Call` event, but then made it a separate event.
Today, there's no need for this complexity, as all the events have a unique interceptor.